rjstires/manager
# Add RenderGuard and implement for NodeBalancer config edit

This PR is for evaluating the concept of a "RenderGuard" component. This component prevents its children from updating unless specified props on itself have changed.

To test:
1. Check out this branch
2. Visit a NodeBalancer Configuration edit page
3. Open the Performance tab of Chrome dev tools, and start recording a profile
4. Tap 3 keys into any of the fields on the page
5. Stop the profile
6. Note the green "frame times" above each of your keypresses at the top of the profile. These are expected to be around 15-30ms with this change.
7. `git checkout 715e9c40b` (RenderGuard is now not implemented)
8. Perform steps 1-6 again. Note that the frame times are now 60-100ms

Also, compare the "highlight updates", with and without RenderGuard on this page.

Is this tedious and error-prone RenderGuard component worth the performance improvement?
